kernel/xpmem_pfn.c

```diff
@@ -388,8 +392,9 @@ xpmem_pin_pages(struct xpmem_thread_group *tg, struct task_struct *src_task,
 	nr_pinned = get_user_pages (src_task, src_mm, vaddr, count, foll_write, 0,
 			      pages, NULL);
 #endif
-	if (!cpumask_empty(&saved_mask))
-		set_cpus_allowed_ptr(current, &saved_mask);
+	if (saved_mask && !cpumask_empty(saved_mask)) {
```

Comment:
seems we don't need to check cpumask_empty now
(and it can be an expensive check)

Reply:
removed, as saved_mask should not be empty